midAssessment/sum_of_digits_of_given_number.rb

```diff
@@ -0,0 +1,20 @@
+# Design a program to calculate the sum of digits of a given integer
+def sum_of_digits(number)
+  sum = 0
+
+  # Loop through each digit
+  while number != 0
+    sum += number % 10  # Add last digit to the sum
+    number /= 10        # Remove the last digit
+  end
+
+  sum
+end
+
+# Example:
+print 'Enter an integer: '
+user_input = gets.chomp.to_i
+
+result = sum_of_digits(user_input)
+
+puts "Sum of digits: #{result}"
```

Comment:
A more optimised solution.
`
number.to_s.chars.map(&:to_i).sum
`

Read about map in ruby